File: src/ContosoAirlinePOS/ContosoAirlinePOSCpp/ContosoAirlinePOSCpp/ContosoAirlinePOSCpp/Booking.cpp
#include "pch.h"
#include "Booking.h"
#include "Booking.g.cpp"

namespace winrt::ContosoAirlinePOSCpp::implementation
{
    hstring Booking::BookingId()
    {
        return m_bookingID;
    }
    void Booking::BookingId(hstring const& value)
    {
        if (m_bookingID != value)
        {
            m_bookingID = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"BookingId" });
        }
    }
    hstring Booking::RegisteredMobileNumber()
    {
        return m_registerMobileNumber;
    }
    void Booking::RegisteredMobileNumber(hstring const& value)
    {
        if (m_registerMobileNumber != value)
        {
            m_registerMobileNumber = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"RegisteredMobileNumber" });
        }
    }
    hstring Booking::RegisteredEmailAddress()
    {
        return m_registerEmailAddress;
    }
    void Booking::RegisteredEmailAddress(hstring const& value)
    {
        if (m_registerEmailAddress != value)
        {
            m_registerEmailAddress = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"RegisteredEmailAddress" });
        }
    }
    hstring Booking::DateOfBooking()
    {
        return m_dateOfBooking;
    }
    void Booking::DateOfBooking(hstring const& value)
    {
        if (m_dateOfBooking != value)
        {
            m_dateOfBooking = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"DateOfBooking" });
        }
    }
    int32_t Booking::FlightId()
    {
        return m_flightID;
    }
    void Booking::FlightId(int32_t value)
    {
        if (m_flightID != value)
        {
            m_flightID = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"FlightId" });
        }
    }
    hstring Booking::TransactionId()
    {
        return m_transactionID;
    }
    void Booking::TransactionId(hstring const& value)
    {
        if (m_transactionID != value)
        {
            m_transactionID = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"TransactionId" });
        }
    }
    float Booking::InvoiceAmount()
    {
        return m_invoiceAmount;
    }
    void Booking::InvoiceAmount(float value)
    {
        UpdateValue<float>(L"InvoiceAmount", m_invoiceAmount, value);
    }
    int32_t Booking::Passengers()
    {
        return m_passengers;
    }
    void Booking::Passengers(int32_t value)
    {
        if (m_passengers != value)
        {
            m_passengers = value;
            m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"Passengers" });
        }
    }
    winrt::event_token Booking::PropertyChanged(Microsoft::UI::Xaml::Data::PropertyChangedEventHandler const& handler)
    {
        return m_propertyChanged.add(handler);
    }
    void Booking::PropertyChanged(winrt::event_token const& token) noexcept
    {
        m_propertyChanged.remove(token);
    }
}

File: src/ContosoAirlinePOS/ContosoAirlinePOSCs/ContosoAirlinePOSCs/ContosoAirlinePOSCs/Models/Booking.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ContosoAirlinePOSCs.Models
{
    public class Booking
    {
        private int _bookingId;
        public int BookingId
        {
            get { return _bookingId; }
            set { _bookingId = value; }
        }

        private string _registeredMobileNumber;
        public string RegisteredMobileNumber
        {
            get { return _registeredMobileNumber; }
            set { _registeredMobileNumber = value; }
        }

        private string _registeredEmailAddress;
        public string RegisteredEmailAddress
        {
            get { return _registeredEmailAddress; }
            set { _registeredEmailAddress = value; }
        }

        private DateTime _dateOfBooking;
        public DateTime DateOfBooking
        {
            get { return _dateOfBooking; }
            set { _dateOfBooking = value; }
        }

        private int _flightId;
        public int FlightId
        {
            get { return _flightId; }
            set { _flightId = value; }
        }

        private List<People> _passengers;
        public List<People> Passengers
        {
            get { return _passengers; }
            set { _passengers = value; }
        }

        private double _invoiceAmount;
        public double InvoiceAmount
        {
            get { return _invoiceAmount; }
            set { _invoiceAmount = value; }
        }

        private string _transactionId;
        public string TransactionId
        {
            get { return _transactionId; }
            set { _transactionId = value; }
        }
    }
}
